src/Elsa.Dashboard/src/components/common-properties/he-question-data-dictionary-property.tsx

```diff
@@ -0,0 +1,120 @@
+import { Component, h, Event, EventEmitter, Prop, State } from '@stencil/core';
+
+import { SyntaxNames } from '../../constants/constants';
+import { DataDictionary, DataDictionaryGroup } from '../../models/custom-component-models';
+import {
+  ActivityDefinitionProperty,
+  ActivityModel,
+  ActivityPropertyDescriptor,
+} from "../../models/elsa-interfaces";
+
+//import { mapSyntaxToLanguage, parseJson, Map } from "../../utils/utils";
+import { parseJson } from "../../utils/utils";
+
+@Component({
+  tag: 'he-question-data-dictionary-property',
+  shadow: false,
+})
+//Copy of Elsa Switch Case
+//Copied to allow us control over how the expression editor is displayed.
+export class HEQuestionDataDictionaryProperty {
+
+  @Prop() activityModel: ActivityModel;
+  @Prop() propertyDescriptor: ActivityPropertyDescriptor;
+  @Prop() propertyModel: ActivityDefinitionProperty;
+  @Event() expressionChanged: EventEmitter<string>;
+  @State() dataDictionary: Array<DataDictionary> = [];
+  @State() selectedDataDictionaryItem: number;
+
+  groupDictionary: Array<DataDictionaryGroup> = [];
+  selectedGroupDictionaryId: number;
+  selectedGroup: DataDictionaryGroup;
+  dataDictionaryDisplayToggle: boolean = false;
+
+  async componentWillLoad() {
+    this.groupDictionary = parseJson(this.propertyDescriptor.options);
+    this.dataDictionary = [];
+    this.selectedDataDictionaryItem = parseInt(this.propertyModel.expressions[SyntaxNames.Literal]);
+  
+    this.selectedGroup = this.groupDictionary.filter(x => x.QuestionDataDictionaryList.filter(y => y.Id == this.selectedDataDictionaryItem)[0])[0];
+    if (this.selectedGroup != undefined) {
+      this.dataDictionaryDisplayToggle = true;
+      this.dataDictionary = [...this.selectedGroup.QuestionDataDictionaryList];
+
+    }
+  }
+
+  updatePropertyModel(updatedValue: string) {
+    this.propertyModel.expressions[SyntaxNames.Literal] = updatedValue;
+    this.expressionChanged.emit(JSON.stringify(this.propertyModel))
+  }
+
+  onGroupDictionaryOptionChanged(e: Event) {
+    const select = e.currentTarget as HTMLSelectElement;
+    this.selectedGroupDictionaryId = parseInt(select.value);
+    let selectedGroup = this.groupDictionary.filter(x => x.Id === this.selectedGroupDictionaryId)[0];
+    if (selectedGroup != undefined) {
+      this.selectedDataDictionaryItem = 0;
+      this.selectedGroup = selectedGroup;
+      this.dataDictionaryDisplayToggle = true;
+      this.dataDictionary = [...selectedGroup.QuestionDataDictionaryList];
+
+    }
+    else {
+      this.dataDictionaryDisplayToggle = false;
+      this.dataDictionary = [];
+    }
+    this.updatePropertyModel('');
```

Comment:
I assume this is to act as a reset for the value, if the user were to change the Group, but not select an option?